File: bestia-models/src/test/java/net/bestia/model/util/ObjectSerializerTest.java
package net.bestia.model.util;

import java.util.Arrays;

import org.junit.Assert;
import org.junit.Test;

public class ObjectSerializerTest {
	
	private ObjectSerializer<String> serializer = new ObjectSerializer<>();
	
	@Test(expected=NullPointerException.class)
	public void deserialize_null_throws() {
		serializer.deserialize(null);
	}
	
	@Test(expected=NullPointerException.class)
	public void serialize_null_throws() {
		serializer.serialize(null);
	}

	@Test
	public void deserialize_invalidData_null() {
		byte[] invalidData = new byte[10];
		Arrays.fill(invalidData, (byte) 0x12);
		
		String test = serializer.deserialize(invalidData);
		Assert.assertNull(test);
	}
	
	@Test
	public void serializeAndDeserialize_validObject_ok() {
		String test = "Hello World";
		byte[] data = serializer.serialize(test);
		
		Assert.assertNotNull(data);
		Assert.assertTrue(data.length > 0);
		
		String test2 = serializer.deserialize(data);
		
		Assert.assertEquals(test, test2);
	}
}

File: bestia-models/src/test/kotlin/net/bestia/model/util/ObjectSerializerTest.kt
package net.bestia.model.util

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import java.util.Arrays

class ObjectSerializerTest {
  @Test
  fun deserialize_invalidData_null() {
    val invalidData = ByteArray(10)
    Arrays.fill(invalidData, 0x12.toByte())

    val test: String? = ObjectSerializer.deserialize(invalidData)
    assertNull(test)
  }

  @Test
  fun serializeAndDeserialize_validObject_ok() {
    val test = "Hello World"
    val data = ObjectSerializer.serialize(test)

    assertNotNull(data)
    assertTrue(data!!.isNotEmpty())

    val test2 = ObjectSerializer.deserialize<String>(data)

    assertEquals(test, test2)
  }
}
